### PR TITLE
Fix some bugs with the inference code

### PR DESCRIPTION
**Pull request by LoryPack**
I fixed two bugs in the source code for: 

-  PMC: it did not properly update the covariance matrix across steps. 
- SMCABC: in case resampling was needed, there was a crash due to trying to index a list with an array of indices. I solved it using list comprehension. 

Further, there are some minor improvements to the plot routine in the Journal class.

### CODE REVIEW

**Comment by LoryPack** [comment]
I've also changed requirements as the fixed version 2.0.0 of glmnet was incompatible with the newer sklearn, as importing from glmnet raised an error. 

Right now, two errors in the tests are still obtained. One of them (in PMCTests), I believe it is due to the fact that it is comparing the PMC result with a precomputed numerical value, which may have been wrong in the first place due to the error in the PMC sampler. 

The other error is instead in PenLogRegTests and is again a numerical error. I don't understand how this comes up as no change to the PenLogReg code was done. I believe it may be introduced due to newer version of the glmnet library, which is used in the logistic regression there.

**Comment by LoryPack** [comment]
The update of covariance matrix in the inference schemes was breaking down when the inference was performed on univariate parameters. It is now fixed.

**Comment by LoryPack** [comment]
I've noted random seeding was not done correctly in the PenLogReg approximate likelihood class, breaking reproducibility. I have fixed that.

**Comment by LoryPack** [comment]
I've introduced two additional fixes: 
- With version 2.2.1 of `glmnet`, the way the cross validation split is performed in LogitNet changed, and shuffling was introduced. Prior versions (2.0.0) instead did not shuffle data. This changes the behavior of LogitNet in the ApproxLhd and Distance PenLogReg. I have used the capability of the new `glmnet` version to provide groups to the fit routine, and provided them in a balanced way, that has the same behavior of older versions. 
- For PenLogReg distance, this also makes sure that the distance between two identical datasets is 0 even if the number of samples in the two datasets is not a multiple of the number of splits in cross validation. This behavior was not correctly obtained before.
- Finally, I have fixed the generation of the reference data in PenLogReg Approx Lhd; specifically, the way the statistics were computed was incorrect. I have also fixed the tests accordingly.

**Comment by LoryPack** [comment]
With the last commit, I have changed the scoring rule that LogitNet uses to pick the best hyperparameter lambda for Lasso logistic regression in PenLogReg approximate likelihood. Specifically, before it was using as scoring the (default) classification accuracy. That however does not take into account the probabilistic prediction of the logistic regression model, which is what instead we are aiming to optimize for. Using `log_loss` scoring in LogitNet does the job (as that is a proper scoring rule) and leads to better performances of the algorithm (see for instance [here](https://www.fharrell.com/post/class-damage/) for some intuitions).
